packages/core/core-flows/src/definition/cart/steps/get-promotion-codes-to-apply.ts

```diff
@@ -0,0 +1,65 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { CartDTO, IPromotionModuleService } from "@medusajs/types"
+import { PromotionActions } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  cart: CartDTO
```

Comment:
```suggestion
  cart: {
    items?: { adjustments?: { code?: string }[] }[]
    shipping_methods?: { adjustments?: { code?: string }[] }[]
  }
```

Reply:
you might be able to reuse some existing types, but the idea stays that we dont really need an entire Cart here but rather a specific data shape 👍